From 8d6e2bb90630469412f2dec5e0ae3a42a26efd20 Mon Sep 17 00:00:00 2001
From: Geoff Thorpe <Geoff.Thorpe@freescale.com>
Date: Wed, 14 Mar 2012 04:49:41 +0000
Subject: [PATCH 075/121] qman: don't automatically dequeue from all
 pool-channels.

Users should add their pool-channels to the cpus/portals they wish to
run with. Otherwise, there is implicitly a race-condition associated
with users needing to *remove* pool-channels from portals that should
not receive dequeues from those pool-channels. The device-tree
portal->pool-channel mapping is supposed to represent which
pool-channels a portal *can* dequeue from, not those which they should
automatically start dequeuing from upon boot-up.

Signed-off-by: Geoff Thorpe <Geoff.Thorpe@freescale.com>
[Kevin: Original patch taken from FSL
QorIQ-SDK-V1.2-SOURCE-20120614-yocto.iso image
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/staging/fsl_qbman/qman_driver.c |    2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/staging/fsl_qbman/qman_driver.c b/drivers/staging/fsl_qbman/qman_driver.c
index 49cdd5a..9e962ba 100644
--- a/drivers/staging/fsl_qbman/qman_driver.c
+++ b/drivers/staging/fsl_qbman/qman_driver.c
@@ -281,8 +281,6 @@ static __init int qman_init(void)
 		p = qman_create_affine_portal(pcfg, NULL);
 		if (p) {
 			u32 irq_sources = 0;
-			/* default: enable all (available) pool channels */
-			qman_static_dequeue_add(~0);
 			/* Determine what should be interrupt-vs-poll driven */
 #ifdef CONFIG_FSL_DPA_PIRQ_SLOW
 			irq_sources |= QM_PIRQ_EQCI | QM_PIRQ_EQRI |
-- 
1.7.9.7

